### PR TITLE
Fixed pagination for Genres, completed Developer Mod & Cont

### PR DESCRIPTION
**Pull request by RoLajoie**
Fixed pagination for Genres, added Controller, Model and Pagination for Developer resource

### CODE REVIEW

**Comment by cr4zy30** [comment]
Update collection and singleton controllers with new validation methods. View handleGetGames() and handleGetGameById()

**Review by YousefAbdelnour — CHANGES_REQUESTED** [review]
Follow Dennis' helpers function. good to go otherwise though

**Review by cr4zy30 — CHANGES_REQUESTED** [review]
Everything looks great, just some small changes:

return $this->renderJson($response, [
"data" => $developer,
], StatusCodeInterface::STATUS_OK);

Some of our current code doesn't follow this practice yet, but we agreed to name the key as the name of the collection being returned, in this case, instead of being "data", it would be "developer" for singletons, and "developers" for collections. When it's a sub-collection, let's say /developer/{developer_id}/games, it would also be "developer" with a list of "games" inside the object.

Also, you can keep it this way because it doesn't change anything, but it's not mandatory to specify the status code, it's 200 OK by default.

👍💯

File: app/src/Controllers/DeveloperController.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        $this->validateIdNum($dev_id, $request, "games");
+
+        $developer = $this->devModel->getDeveloperById($dev_id);
+
+        $this->validateObj($developer, $request, "Could not find game with id [{$dev_id}]");
+
+        //render Json and send the data
+        return $this->renderJson($response, [
+            "data" => $developer,
```
Comment (YousefAbdelnour) [inline app/src/Controllers/DeveloperController.php:53]: "data" supposed to be "developer" or something like that. follow naming convention

File: app/src/Controllers/DeveloperController.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        $params = $request->getQueryParams();
+
+        //using custom method to validate parameters
+        $this->devModel->setPaginationOptions($this->getValidatedPaginationParams($params, $request));
+
+        $developers = $this->devModel->getDevelopers($params);
+
+        //renderJson and send the data
+        return $this->renderJson($response, [
+            "data" => $developers
```
Comment (YousefAbdelnour) [inline app/src/Controllers/DeveloperController.php:34]: "data" supposed to be "developer" or something like that. follow naming convention

**Review by cr4zy30 — APPROVED** [review]
"data" => $developers

But that's fine you can change it next PR

**Review by YousefAbdelnour — CHANGES_REQUESTED** [review]

**Comment by YousefAbdelnour** [comment]
If you want to merge and then change that for the next PR. fine by me

Approval up to you @cr4zy30 

(edit) sorry didn' see your approval Denis

**Pull request merged**